clusterlib/tests/test_scheduler.py

```diff
@@ -61,6 +61,56 @@ def test_fixed_backend():
     assert_raises(ValueError, _get_backend, 'hadoop')
 
 
+def test_queued_or_running_jobs_no_backend():
+
+    if (_which('qsub') is not None and
+            _which('sbatch') is not None):
+        raise SkipTest("This platform has either slurm or sge installed")
+
+    assert_equal(queued_or_running_jobs(), [])
+
+
+def test_queued_or_running_jobs_sge():
+    """Test queued or running job function on sge"""
+    # Check that slurm is installed
+    if _which('qsub') is None:
+        raise SkipTest("qsub (sge) is missing")
+
+    user = getuser()
+    job_name = "test-sleepy-job"
+
+    # Launch a sleepy slurm job
+    sleep_job = submit(job_command="sleep 600", job_name=job_name,
+                       backend="sge", time="700", memory=100)
+    os.system(sleep_job)
+
+   # Get job id
+    job_id = None
+    try:
+        out = subprocess.check_output("qstat -xml -u {0}".format(user),
```

Comment:
BTW, it's better to use the `check_output(["qstat", "-xml", "-u", user, "-j", job_name])` to avoid shell injection security holes.